internal/utils/utils.go

```diff
@@ -27,3 +28,7 @@ func UpdateConditions(existingConditions []metav1.Condition, newCondition metav1
 
 	return append([]metav1.Condition{newCondition}, existingConditions...)
 }
+
+func IsAllowedNamespaces(namespace string, namespaces []string) bool {
+	return glob.MatchStringInList(namespaces, namespace, glob.REGEXP)
```

Comment:
Please avoid regex if possible: https://owasp.org/www-community/attacks/Regular_expression_Denial_of_Service_-_ReDoS

I think wildcard prefix/suffix would be enough here:

https://gitlab.com/kubitus-project/grafana-org-operator/-/blob/dc659439102018722ed5f757554f164100af2264/api/v1beta1/types.go#L97-108